test/WebSites/ControllersFromServicesWebSite/Startup.cs

```diff
@@ -25,10 +25,15 @@ public void ConfigureServices(IServiceCollection services)
                 .AddMvc()
                 .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Clear())
                 .AddApplicationPart(typeof(TimeScheduleController).GetTypeInfo().Assembly)
-                .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Add(new TypesPart(
-                    typeof(AnotherController),
-                    typeof(ComponentFromServicesViewComponent),
-                    typeof(InServicesTagHelper))))
+                .ConfigureApplicationPartManager(manager =>
+                {
+                    manager.ApplicationParts.Add(new TypesPart(
+                      typeof(AnotherController),
+                      typeof(ComponentFromServicesViewComponent),
+                      typeof(InServicesTagHelper)));
+
+                    manager.FeatureProviders.Add(new AssemblyMetadataReferenceFeatureProvider());
```

Comment:
I would add preserve compilation context always, what's the cost, size on the assembly? Adding just the assembly i'm not sure if it'll work. However, for apps that don't load assemblies dinamically I believe that everything will work as long as the top level assembly has preserveCompilationContext = true, as it will contain the transitive closure of all the dependencies.